### ISSUE TITLE
Error when interpolating DistributedFEFunction

### ISSUE THREAD

**Issue by oriolcg** [gridap/GridapDistributed.jl#69]
Hi @amartinhuertas , @fverdugo,

I'm facing an error when interpolating a DistributedFEFunction. The `interpolate` function works well when the input function is a `Function`, but it gives an error for a FE function defined in the same FE space. Do you have any idea on why this is happening and how to solve it? Thank you!

This is a reproducer:
```julia
using Gridap
using GridapDistributed
using PartitionedArrays

parts = get_part_ids(sequential,(1,))
domain = (0,1)
cells = (2,)
𝒯 = CartesianDiscreteModel(parts,domain,cells)
refFE = ReferenceFE(lagrangian,Float64,1)
V = TestFESpace(𝒯,refFE)
U = TrialFESpace(V)
u₀(x) = 1.0
uₕ₀ = interpolate(u₀,U)
uₕ = interpolate(uₕ₀,U)
```

And this is the error
```
ERROR: LoadError: MethodError: no method matching Gridap.CellData.CellField(::GridapDistributed.DistributedCellField{PartitionedArrays.SequentialData{Gridap.FESpaces.SingleFieldFEFunction{Gridap.CellData.GenericCellField{Gridap.CellData.ReferenceDomain}}, 1}, GridapDistributed.DistributedFEFunctionData{PartitionedArrays.PVector{Float64, PartitionedArrays.SequentialData{Vector{Float64}, 1}, PartitionedArrays.PRange{PartitionedArrays.SequentialData{PartitionedArrays.IndexSet, 1}, PartitionedArrays.Exchanger{PartitionedArrays.SequentialData{Vector{Int32}, 1}, PartitionedArrays.SequentialData{PartitionedArrays.Table{Int32}, 1}}, Nothing}}}}, ::Gridap.Geometry.BodyFittedTriangulation{1, 1, Gridap.Geometry.CartesianDiscreteModel{1, Float64, typeof(identity)}, Gridap.Geometry.CartesianGrid{1, Float64, typeof(identity)}, Gridap.Arrays.IdentityVector{Int64}}, ::Gridap.CellData.ReferenceDomain)
Closest candidates are:
  Gridap.CellData.CellField(::AbstractArray{<:Number}, ::Gridap.Geometry.Triangulation, ::Gridap.CellData.DomainStyle) at C:\Users\ocolomesgene\.julia\packages\Gridap\775Gn\src\CellData\CellFields.jl:89
  Gridap.CellData.CellField(::Number, ::Gridap.Geometry.Triangulation, ::Gridap.CellData.DomainStyle) at C:\Users\ocolomesgene\.julia\packages\Gridap\775Gn\src\CellData\CellFields.jl:83
  Gridap.CellData.CellField(::Function, ::Gridap.Geometry.Triangulation, ::Gridap.CellData.DomainStyle) at C:\Users\ocolomesgene\.julia\packages\Gridap\775Gn\src\CellData\CellFields.jl:77
  ...
Stacktrace:
  [1] _cell_vals(fs::Gridap.FESpaces.UnconstrainedFESpace{Vector{Float64}, Gridap.FESpaces.NodeToDofGlue{Int32}}, object::GridapDistributed.DistributedCellField{PartitionedArrays.SequentialData{Gridap.FESpaces.SingleFieldFEFunction{Gridap.CellData.GenericCellField{Gridap.CellData.ReferenceDomain}}, 1}, GridapDistributed.DistributedFEFunctionData{PartitionedArrays.PVector{Float64, PartitionedArrays.SequentialData{Vector{Float64}, 1}, PartitionedArrays.PRange{PartitionedArrays.SequentialData{PartitionedArrays.IndexSet, 1}, PartitionedArrays.Exchanger{PartitionedArrays.SequentialData{Vector{Int32}, 1}, PartitionedArrays.SequentialData{PartitionedArrays.Table{Int32}, 1}}, Nothing}}}})
    @ Gridap.FESpaces C:\Users\ocolomesgene\.julia\packages\Gridap\775Gn\src\FESpaces\SingleFieldFESpaces.jl:193
  [2] interpolate!(object::GridapDistributed.DistributedCellField{PartitionedArrays.SequentialData{Gridap.FESpaces.SingleFieldFEFunction{Gridap.CellData.GenericCellField{Gridap.CellData.ReferenceDomain}}, 1}, GridapDistributed.DistributedFEFunctionData{PartitionedArrays.PVector{Float64, PartitionedArrays.SequentialData{Vector{Float64}, 1}, PartitionedArrays.PRange{PartitionedArrays.SequentialData{PartitionedArrays.IndexSet, 1}, PartitionedArrays.Exchanger{PartitionedArrays.SequentialData{Vector{Int32}, 1}, PartitionedArrays.SequentialData{PartitionedArrays.Table{Int32}, 1}}, Nothing}}}}, free_values::Vector{Float64}, fs::Gridap.FESpaces.UnconstrainedFESpace{Vector{Float64}, Gridap.FESpaces.NodeToDofGlue{Int32}})
    @ Gridap.FESpaces C:\Users\ocolomesgene\.julia\packages\Gridap\775Gn\src\FESpaces\SingleFieldFESpaces.jl:185
  [3] (::GridapDistributed.var"#158#159"{GridapDistributed.DistributedCellField{PartitionedArrays.SequentialData{Gridap.FESpaces.SingleFieldFEFunction{Gridap.CellData.GenericCellField{Gridap.CellData.ReferenceDomain}}, 1}, GridapDistributed.DistributedFEFunctionData{PartitionedArrays.PVector{Float64, PartitionedArrays.SequentialData{Vector{Float64}, 1}, PartitionedArrays.PRange{PartitionedArrays.SequentialData{PartitionedArrays.IndexSet, 1}, PartitionedArrays.Exchanger{PartitionedArrays.SequentialData{Vector{Int32}, 1}, PartitionedArrays.SequentialData{PartitionedArrays.Table{Int32}, 1}}, Nothing}}}}})(V::Gridap.FESpaces.UnconstrainedFESpace{Vector{Float64}, Gridap.FESpaces.NodeToDofGlue{Int32}}, vec::Vector{Float64})
    @ GridapDistributed C:\Users\ocolomesgene\Progs\git-repos\gridap\GridapDistributed.jl\src\FESpaces.jl:356
  [4] (::Base.var"#4#5"{GridapDistributed.var"#158#159"{GridapDistributed.DistributedCellField{PartitionedArrays.SequentialData{Gridap.FESpaces.SingleFieldFEFunction{Gridap.CellData.GenericCellField{Gridap.CellData.ReferenceDomain}}, 1}, GridapDistributed.DistributedFEFunctionData{PartitionedArrays.PVector{Float64, PartitionedArrays.SequentialData{Vector{Float64}, 1}, PartitionedArrays.PRange{PartitionedArrays.SequentialData{PartitionedArrays.IndexSet, 1}, PartitionedArrays.Exchanger{PartitionedArrays.SequentialData{Vector{Int32}, 1}, PartitionedArrays.SequentialData{PartitionedArrays.Table{Int32}, 1}}, Nothing}}}}}})(a::Tuple{Gridap.FESpaces.UnconstrainedFESpace{Vector{Float64}, Gridap.FESpaces.NodeToDofGlue{Int32}}, Vector{Float64}})
    @ Base .\generator.jl:36
  [5] iterate
    @ .\generator.jl:47 [inlined]
  [6] collect(itr::Base.Generator{Base.Iterators.Zip{Tuple{Vector{Gridap.FESpaces.UnconstrainedFESpace{Vector{Float64}, Gridap.FESpaces.NodeToDofGlue{Int32}}}, Vector{Vector{Float64}}}}, Base.var"#4#5"{GridapDistributed.var"#158#159"{GridapDistributed.DistributedCellField{PartitionedArrays.SequentialData{Gridap.FESpaces.SingleFieldFEFunction{Gridap.CellData.GenericCellField{Gridap.CellData.ReferenceDomain}}, 1}, GridapDistributed.DistributedFEFunctionData{PartitionedArrays.PVector{Float64, PartitionedArrays.SequentialData{Vector{Float64}, 1}, PartitionedArrays.PRange{PartitionedArrays.SequentialData{PartitionedArrays.IndexSet, 1}, PartitionedArrays.Exchanger{PartitionedArrays.SequentialData{Vector{Int32}, 1}, PartitionedArrays.SequentialData{PartitionedArrays.Table{Int32}, 1}}, Nothing}}}}}}})
    @ Base .\array.jl:724
  [7] map(::Function, ::Vector{Gridap.FESpaces.UnconstrainedFESpace{Vector{Float64}, Gridap.FESpaces.NodeToDofGlue{Int32}}}, ::Vector{Vector{Float64}})
    @ Base .\abstractarray.jl:2948
  [8] map_parts(::Function, ::PartitionedArrays.SequentialData{Gridap.FESpaces.UnconstrainedFESpace{Vector{Float64}, Gridap.FESpaces.NodeToDofGlue{Int32}}, 1}, ::Vararg{PartitionedArrays.SequentialData})
    @ PartitionedArrays C:\Users\ocolomesgene\Progs\git-repos\gridap\PartitionedArrays.jl\src\SequentialBackend.jl:56
  [9] interpolate!
    @ C:\Users\ocolomesgene\Progs\git-repos\gridap\GridapDistributed.jl\src\FESpaces.jl:355 [inlined]
 [10] interpolate(u::GridapDistributed.DistributedCellField{PartitionedArrays.SequentialData{Gridap.FESpaces.SingleFieldFEFunction{Gridap.CellData.GenericCellField{Gridap.CellData.ReferenceDomain}}, 1}, GridapDistributed.DistributedFEFunctionData{PartitionedArrays.PVector{Float64, PartitionedArrays.SequentialData{Vector{Float64}, 1}, PartitionedArrays.PRange{PartitionedArrays.SequentialData{PartitionedArrays.IndexSet, 1}, PartitionedArrays.Exchanger{PartitionedArrays.SequentialData{Vector{Int32}, 1}, PartitionedArrays.SequentialData{PartitionedArrays.Table{Int32}, 1}}, Nothing}}}}, f::GridapDistributed.DistributedSingleFieldFESpace{PartitionedArrays.SequentialData{Gridap.FESpaces.UnconstrainedFESpace{Vector{Float64}, Gridap.FESpaces.NodeToDofGlue{Int32}}, 1}, PartitionedArrays.PRange{PartitionedArrays.SequentialData{PartitionedArrays.IndexSet, 1}, PartitionedArrays.Exchanger{PartitionedArrays.SequentialData{Vector{Int32}, 1}, PartitionedArrays.SequentialData{PartitionedArrays.Table{Int32}, 1}}, Nothing}, PartitionedArrays.PVector{Float64, PartitionedArrays.SequentialData{Vector{Float64}, 1}, PartitionedArrays.PRange{PartitionedArrays.SequentialData{PartitionedArrays.IndexSet, 1}, PartitionedArrays.Exchanger{PartitionedArrays.SequentialData{Vector{Int32}, 1}, PartitionedArrays.SequentialData{PartitionedArrays.Table{Int32}, 1}}, Nothing}}})
    @ GridapDistributed C:\Users\ocolomesgene\Progs\git-repos\gridap\GridapDistributed.jl\src\FESpaces.jl:350
```

**Comment by oriolcg** [comment]
Just as a remark, this is not a critical issue since the interpolation between FE functions in the same space can be just done with the `FEFunction` constructor:
```julia
uₕ = FEFunction(U,get_free_dof_values(uₕ₀))
```

**Comment by amartinhuertas** [comment]
Hi @oriolcg, thanks for reporting. I will take a look when I have some gap and come back to you.

**Comment by oriolcg** [comment]
This problem is simply caused by the fact that the `CellField` function is not implemented for `DistributedCellField`. The equivalent function in gridap is: https://github.com/gridap/Gridap.jl/blob/dc44108dfab4e8c581501e7cce98fa1ca75324d0/src/CellData/CellFields.jl#L99. Is there any particular reason why this one is not in `GridapDistributed`?

**Comment by oriolcg** [comment]
Issue fixed in commit 9a3a8b845fa0d7e4638448162dff68588aad9a2f. I just need to add tests and I'll create a PR.
